src/modules-load/modules-load.c

```diff
@@ -163,34 +179,44 @@ static int run(int argc, char *argv[]) {
         if (r < 0)
                 log_warning_errno(r, "Failed to parse kernel command line, ignoring: %m");
 
-        r = module_setup_context(&ctx);
-        if (r < 0)
-                return log_error_errno(r, "Failed to initialize libkmod context: %m");
+        int sockets[2];
+        const int max_tasks = MAX(MIN(sysconf(_SC_NPROCESSORS_ONLN) / 2, 10), 2);
+        pthread_t *threads = malloc(sizeof(pthread_t) * max_tasks);
+        if (socketpair(AF_UNIX, SOCK_SEQPACKET, 0, sockets) == -1)
+                log_error_errno(errno, "socketpair");
 
-        r = 0;
+        // Create worker threads
+        for (int i = 0; i < max_tasks; ++i)
+                if (pthread_create(&threads[i], NULL, worker_thread, &sockets[1]) != 0)
+                        log_error_errno(errno, "pthread_create");
```

Comment:
pthread apis are weird, they return a possitive error code directly as func return, and do not set `errno`.

also, this should be fatal.

(or maybe it should only be fatal in case if i == 0, and otherwise just write a log message and exit the loop, but continue. This might be nice so that people can do TasksMax=4 in the unit file and really nicely put a limit on worker threads we fork